services/azure-devops/azure-devops-build.service.js

```diff
@@ -58,22 +65,60 @@ module.exports = class AzureDevOpsBuild extends BaseSvgScrapingService {
         pattern: ':organization/:projectId/:definitionId/:branch',
         namedParams: {
           organization: 'totodem',
-          projectId: '8cf3ec0e-d0c2-4fcd-8206-ad204f254a96',
+          projectId: 'Shields.io',
           definitionId: '2',
           branch: 'master',
         },
         staticPreview: renderBuildStatusBadge({ status: 'succeeded' }),
         keywords,
         documentation,
       },
+      {
+        title: 'Azure DevOps builds (stage)',
+        pattern:
+          ':organization/:projectId/:definitionId/:branch*\\?stage=:stage',
+        namedParams: {
+          organization: 'totodem',
+          projectId: 'Shields.io',
+          definitionId: '5',
+          branch: 'master',
+          stage: 'Successful Stage',
+        },
+        staticPreview: renderBuildStatusBadge({ status: 'succeeded' }),
+        keywords,
+        documentation,
+      },
+      {
+        title: 'Azure DevOps builds (job)',
+        pattern:
+          ':organization/:projectId/:definitionId/:branch*\\?stage=:stage&job=:job',
+        namedParams: {
+          organization: 'totodem',
+          projectId: 'Shields.io',
+          definitionId: '5',
+          branch: 'master',
+          stage: 'Successful Stage',
+          job: 'Successful Job',
+        },
+        staticPreview: renderBuildStatusBadge({ status: 'succeeded' }),
+        keywords,
+        documentation,
+      },
     ]
   }
 
-  async handle({ organization, projectId, definitionId, branch }) {
+  async handle(
+    { organization, projectId, definitionId, branch },
+    { stage, job }
+  ) {
     // Microsoft documentation: https://docs.microsoft.com/en-us/rest/api/vsts/build/status/get
     const { status } = await fetch(this, {
       url: `https://dev.azure.com/${organization}/${projectId}/_apis/build/status/${definitionId}`,
-      qs: { branchName: branch },
+      qs: {
+        branchName: branch,
+        stageName: stage,
+        jobName: job,
```

Comment:
Given that `stageName` must be specified in order to use `jobName`, maybe it's worth validating against that? The Joi schema can probably support this kind of logic, but an explicit check throwing a `InvalidParameter` with a message along the lines of "stage must be specified with job" would do the trick as well!

Reply:
There is nothing preventing from getting a badge for a given job without specifying its stage in Azure DevOps API. I tested it and it works perfectly. That's why I did not implement any validation for that.

Reply:
You're right, I had read the documentation a bit too fast! You only have to specify stage when using job if there a more than one stages: "If there are more than one, then it is required to pass in a stageName value when specifying a jobName".